### ISSUE TITLE
Game crashes on exit.

### ISSUE THREAD

**Issue by tiripon** [BepInEx/BepInEx#290]
### What happened?

Since I installed BepInEx 5.4.14 the gema crashes on exit.

I tried removing every Mods and it still crashes on exit.

It is still 100% playable.

### Steps to reproduce

Launch the game through Steam. Once the game is up click "Quit"

### BepInEx Distribution

Third party distribution

### Log outputs

```txt
[Message:   BepInEx] BepInEx 5.4.14.0 - valheim (7/1/2021 12:19:23 PM)
[Info   :   BepInEx] Running under Unity v2019.4.24.6137584
[Info   :   BepInEx] CLR runtime version: 4.0.30319.42000
[Info   :   BepInEx] Supports SRE: True
[Info   :   BepInEx] System platform: Bits64, Windows
[Message:   BepInEx] Preloader started
[Info   :   BepInEx] Loaded 1 patcher method from [BepInEx.Preloader 5.4.14.0]
[Info   :   BepInEx] 1 patcher plugin loaded
[Info   :   BepInEx] Patching [UnityEngine.CoreModule] with [BepInEx.Chainloader]
[Message:   BepInEx] Preloader finished
Fossilize INFO: Overriding serialization path: "D:\Steam\steamapps\shadercache\892970\fozpipelinesv5\steamapprun_pipeline_cache".
[Message:   BepInEx] Chainloader ready
[Message:   BepInEx] Chainloader started
[Info   :   BepInEx] 1 plugins to load
[Info   :   BepInEx] Loading [Display BepInEx Info In-Game 1.1.0]
[Message:   BepInEx] Chainloader startup complete
[Info   : Unity Log] 08/03/2021 15:42:22: Starting to load scene:start

[Warning: Unity Log] Only custom filters can be played. Please add a custom filter or an audioclip to the audiosource (Amb_MainMenu).
[Info   : Unity Log] 08/03/2021 15:42:25: Valheim version:0.156.2

[Info   : Unity Log] 08/03/2021 15:42:25: Render threading mode:MultiThreaded

[Info   : Unity Log] 08/03/2021 15:42:25: Using environment steamid 892970

[Info   : Unity Log] 08/03/2021 15:42:25: Using steam APPID:892970

Setting breakpad minidump AppID = 892970
SteamInternal_SetMinidumpSteamID:  Caching Steam ID:  76561198043784644 [API loaded no]
[Info   : Unity Log] 08/03/2021 15:42:26: Authentication:k_ESteamNetworkingAvailability_Attempting

[Info   : Unity Log] 08/03/2021 15:42:26: Steam initialized, persona:tiripon

[Warning: Unity Log] 08/03/2021 15:42:26: Missing audio clip in music respawn

[Info   : Unity Log] 08/03/2021 15:42:26: Checking for installed DLCs

[Info   : Unity Log] 08/03/2021 15:42:26: DLC:beta installed:False

[Info   : Unity Log] 08/03/2021 15:42:26: Loaded localization CSV:localization language:English

[Info   : Unity Log] 08/03/2021 15:42:26: Failed to load language file localization_extra

[Info   : Unity Log] 08/03/2021 15:42:26: Loaded localization CSV:localization language:English

[Info   : Unity Log] 08/03/2021 15:42:26: Failed to load language file localization_extra

[Info   : Unity Log] 08/03/2021 15:42:29: UI Group status changed Menu = True

[Info   : Unity Log] 08/03/2021 15:42:48: Steam manager on destroy
```

### Environment

```markdown
- OS: Windows 10
- BepInEx: 5.4.14
- Game: Valheim
```

**Comment by ghorsington** [comment]
Greetings!

Thanks for the information. I edited your ticket a little since from the logs you appear to use [BepInEx from Thunderstore](https://valheim.thunderstore.io/package/denikson/BepInExPack_Valheim/) which is not the stable build from GitHub.

Nevertheless, I am able to replicate the issue with BepInEx 5.4.14 off GitHub as well. This looks like an issue with MonoMod library which got updated in 5.4.14. I will investigate this further but as you said, the game is still playable for the time being.

**Comment by tiripon** [comment]
Thanks! I'm glad to help.
Yes, I used BepInEx from Thunderstore. I thought that this download was from the same version on GitHub.
Thanks for the clarification.
I will download the GitHub version from now on.

**Comment by ghorsington** [comment]
> I will download the GitHub version from now on.

For Valheim, I suggest downloading the builds off Thunderstore since it includes some libraries needed to run BepInEx on Valheim (as Valheim requires some special handling, see #246 if you're interested in more info). You can try using official releases off GitHub, but you'd have to edit the default config a bit yourself then.

As far as issue tickets go, it is primarily important to know exactly which version you downloaded. Some external distributions might not be up-to-date or contain some additional libraries. It is not indicative of a bad build. It's just not the build off GitHub.

**Comment by tiripon** [comment]
Gotcha :) Thanks again!

**Comment by ghorsington** [comment]
This is now fixed in 5.4.15. The same update has also been pushed to Thunderstore, please update when possible.

**Comment by tiripon** [comment]
Yes, I can confirm that it is fixed :)

Thank you!
